packages/tasks/src/model-libraries-snippets.ts

```
@@ -394,32 +394,91 @@ model = keras.saving.load_model("hf://${model.id}")
 `,
 ];
 
-export const keras_nlp = (model: ModelData): string[] => [
-	`# Available backend options are: "jax", "torch", "tensorflow".
-import os
-os.environ["KERAS_BACKEND"] = "jax"
+const _keras_hub_causal_lm = (modelId: string): string => `
+import keras_hub
 
-import keras_nlp
+# Load CausalLM model (optional: use half precision for inference)
+causal_lm = keras_hub.models.CausalLM.from_preset(${modelId}, dtype="bfloat16")
+causal_lm.compile(sampler="greedy")  # (optional) specify a sampler
 
-tokenizer = keras_nlp.models.Tokenizer.from_preset("hf://${model.id}")
-backbone = keras_nlp.models.Backbone.from_preset("hf://${model.id}")
-`,
-];
+# Generate text
+causal_lm.generate("Keras: deep learning for", max_length=64)
+`;
 
-export const keras_hub = (model: ModelData): string[] => [
-	`# Available backend options are: "jax", "torch", "tensorflow".
-import os
-os.environ["KERAS_BACKEND"] = "jax"
+const _keras_hub_text_to_image = (modelId: string): string => `
+import keras_hub
+
+# Load TextToImage model (optional: use half precision for inference)
+text_to_image = keras_hub.models.TextToImage.from_preset(${modelId}, dtype="bfloat16")
+
+# Generate images with a TextToImage model.
+text_to_image.generate("Astronaut in a jungle")
+`;
 
+const _keras_hub_text_classifier = (modelId: string): string => `
 import keras_hub
 
-# Load a task-specific model (*replace CausalLM with your task*)
-model = keras_hub.models.CausalLM.from_preset("hf://${model.id}", dtype="bfloat16")
+# Load TextClassifier model
```

Because our most popular classifiers won't have pretrained heads (thinking of bert likes). I think this makes sense to show with some toy fine-tuning. E.g.

```python
import keras_hub
import keras

# Load TextClassifier model
text_classifier = keras_hub.models.TextClassifier.from_preset(
    ${modelId},
    num_classes=2,
)
# Fine-tune
text_classifier.fit(x=["Thilling adventure!", "Total snoozefest."], y=[1, 0])
# Classify text
text_classifier.predict(["Not my cup of tea."])
```

---

Is there a reason why we need to provide `num_classes=2` ? Shouldn't this depend on the model we chose ?

I feel that we should aim at showcasing how to use the model on the specific task it has been trained on (i.e. the simplest use case). If we add a `num_classes=2` + a `.fit(...)` step, then a normal user won't be able to just use the model. They would need to know they have to remove the `num_classes` parameter.

---

@mattdangerw  If we want to show how to finetune + other use cases, what do you think about linking to the base class documentation page like this?

```py
# Check out https://keras.io/keras_hub/api/base_classes/text_classifier/ for more examples.
import keras_hub

# Load TextClassifier model
text_classifier = keras_hub.models.TextClassifier.from_preset(${modelId})

# Classify text
text_classifier.predict("Keras is a multi-backend ML framework.")
```

---

no super strong opinion but in our experience it's good to keep snippets as minimal as possible (and focus on predict rather than fit)

Your call in the end though!

---

Thanks!

The issue is really with the bert-likes (but those are fairly prominent in quick starts, etc). Since these have no built in head, if you try to create those without specifying `num_classes` you actually get an error. Something like...

```python
>>> keras_hub.models.TextClassifier.from_preset("roberta_base_en")
Error you must supply num_classes to the classifier
```

I know the transformer answer here is to default these as "fill mask" models. And we could consider something like that but I think it's probably better done as a follow up. Almost everyone will be using these models for classification, and the code I'm showing above will work generally for classifiers (unlike the code in the PR currently).

Let me know if that makes sense! We could also ditch the `fit()` line, but we'd be showing random predictions because the heads won't be trained.

---

Makes perfect sense! Then we'll go with what you've suggested, e.g. num_classes + .fit example. Thanks for the explanation :)

---

updated in https://github.com/huggingface/huggingface.js/pull/1118/commits/e608ac0b8a7927271a7f176ad46a0bd01ec18ca0